kubectl-cloudflow/cmd/deploy.go

```diff
@@ -81,32 +105,198 @@ the stored credentials.
+		PreRun: func(cmd *cobra.Command, args []string) {
+			if cmd.Flag("image").Changed == false && cmd.Flag("blueprint").Changed == false {
+				util.LogAndExit("Please specify either the blueprint file or the image path")
+			}
+			if cmd.Flag("image").Changed == true {
+				fmt.Printf("Deploying image %s\n", cmd.Flag("image").Value.String())
+			} else if cmd.Flag("blueprint").Changed == true {
+				fmt.Printf("Deploying blueprint %s\n", cmd.Flag("blueprint").Value.String())
+			}
+		},
+		Run: deployOpts.deployImpl,
 	}
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.blueprintFilePath, "blueprint", "b", "", "blueprint file path.")
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.imagePath, "image", "i", "", "docker image path.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.username, "username", "u", "", "docker registry username.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.password, "password", "p", "", "docker registry password.")
 	deployOpts.cmd.Flags().BoolVarP(&deployOpts.passwordStdin, "password-stdin", "", false, "Take the password from stdin")
 	deployOpts.cmd.Flags().StringArrayVar(&deployOpts.volumeMounts, "volume-mount", []string{}, "Accepts a key/value pair separated by an equal sign. The key should be the name of the volume mount, specified as '[streamlet-name].[volume-mount-name]'. The value should be the name of an existing persistent volume claim.")
+	deployOpts.cmd.Flags().StringToIntVar(&deployOpts.replicas, "scale", map[string]int{}, "Accepts key/value pairs for replicas per streamlet")
 
 	rootCmd.AddCommand(deployOpts.cmd)
 }
 
 func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 	version.FailOnProtocolVersionMismatch()
 
-	imageRef := args[0]
+	if (cmd.Flag("image").Changed == true) && (cmd.Flag("blueprint").Changed == true) {
+		util.LogAndExit("%s", "Cannot specify both blueprint and image in command")
+	}
+	if (cmd.Flag("image").Changed == false) && (cmd.Flag("blueprint").Changed == false) {
+		util.LogAndExit("%s", "Need to specify either blueprint or image in command")
+	}
+	// if (cmd.Flag("image").Changed == true) && (cmd.Flag("scale").Changed == true) {
+	// util.LogAndExit("%s", "scale is supported only with blueprint based deployment")
+	// }
+	if cmd.Flag("image").Changed == true {
+		deployImage(cmd, opts, args)
+	} else {
+		deployBlueprint(cmd, opts, args)
+	}
+}
+
+func deployBlueprint(cmd *cobra.Command, opts *deployOptions, args []string) {
+	// read blueprint file and load its content
+	blueprintFile := cmd.Flag("blueprint").Value.String()
+	content, err := ioutil.ReadFile(blueprintFile)
+	if err != nil {
+		util.LogAndExit("%s", err.Error())
+	}
+	config, err := loadBlueprintConfigContent(string(content[:]))
+	if err != nil {
+		util.LogAndExit("%s", err.Error())
+	}
+	fmt.Printf("%s\n", config)
+
+	// get the images from the blueprint
+	images := config.GetNode("blueprint.images").GetObject().Items()
```

Comment:
What if it has no "images" (what error will the user see)?

Reply:
@RayRoestenburg  This is the section of code for deploying using a blueprint. And I am assuming that blueprint verification will be done before deploy. This section `images` is a mandatory part of the blueprint configuration. Can we assume that it _will be_ there ?

Reply:
@debasishg @RayRoestenburg I think we should run the verification process before deployment anyway. The reason you want to have `verify` as a standalone command is because you want the user to be able to verify his blueprint while he constructs them or generates them.

Reply:
correct deploy must always verify. That's not here yet cause debasish needs to merge with your code, I guess.

Reply:
exactly .. will be there once verify code is in place. That's why I suggest @skonto 's PR to be merged first and then mine @RayRoestenburg